product-service/handlers/getProductById.ts

```diff
@@ -1,24 +1,33 @@
 import { APIGatewayProxyHandler } from 'aws-lambda';
+import { Client } from 'pg';
 import 'source-map-support/register';
 
-import { products } from '../mock-data/products';
+import { headers } from '../constants/headers';
+import { dbOptions } from '../constants/dbOptions';
 
 export const getProductById: APIGatewayProxyHandler = async (event, _context) => {
-  const { productId } = event.pathParameters;
-  const product = products.find((product) => product.id === productId);
-
-  const statusCode = !!product ? 200 : 404;
-  const body = !!product ? product : {
-    message: "product not found"
-  };
-
-  return {
-    statusCode,
-    headers: {
-      "Access-Control-Allow-Headers" : "Content-Type",
-      "Access-Control-Allow-Origin": "*",
-      "Access-Control-Allow-Methods": "OPTIONS,GET"
-    },
-    body: JSON.stringify(body)
-  };
+  const client = new Client(dbOptions);
+  await client.connect();
+  console.log('Event', event);
+  try {
+    const { productId } = event.pathParameters;
+    const { rows }  = await client.query('select * from products p left join stocks s on p.id = s.product_id where p.id = ($1)', [productId])
```

Comment:
`product_id` field will be returned as well, but according to the task the following structure should be:
```
  {
    id: '19ba3d6a-f8ed-491b-a192-0a33b71b38c4',
    count: 2
    price: 200,
    title: ‘Product Title’,
    description: ‘This product ...’
  }
````
You can extract only fields that are necessary: 
'select p.id, p.title, p.description, p.price, s.count from products p left join stocks s on p.id = s.product_id where p.id = ($1)

Reply:
In addition, it would be better to use 404 status code if there is no corresponding product in DB.

Reply:
+1 to both suggestions from Maryia

Reply:
Thanks for suggestions, pushed the changes